src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -2787,20 +2805,61 @@ int wm_vuldet_insert(wm_vuldet_db *parsed_oval, update_node *update) {
             sqlite3_bind_int(stmt, 7, 0);
             sqlite3_bind_int(stmt, 8, vul_it->ignore);
             sqlite3_bind_int(stmt, 9, 0);
+            if (vul_it->deps) {
+                // Check the dependency array to avoid inserting more IDs than necessary
+                if (deps_size != vul_it->deps->elements) {
+                    deps_modified = true;
+                } else {
+                    for (int i = 0; i < deps_size; ++i) {
+                        if (dependency[i] != vul_it->deps->test_ref[i]) {
+                            deps_modified = true;
+                            break;
+                        }
+                    }
+                }
+                if(deps_modified) {
+                    os_free(dependency);
```

Comment:
If `dependency` is a `char **` it cannot be freed as a simple `char *`. You should free its elements as well. Please review all the uses of `dependency` to ensure no memory leaks or double frees appear.